### PR TITLE
feat(Lazada): add presence

### PR DESCRIPTION
**Pull request by ZzzSleepyy**
## Description 

Adding a new presence for Lazada
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/L/Lazada/metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.9",
+	"author": {
+		"id": "392970361047613441",
+		"name": "niqowi"
+	},
+	"service": "Lazada",
+	"description": {
+		"en": "Visit Lazada.com to get incredible bargains and stacking coupons for even more savings."
+	},
+	"url": "lazada.com",
```
Comment (theusaf) [inline websites/L/Lazada/metadata.json:11]: Site supports `www` subdomain
```suggestion
	"url": [
		"lazada.com",
		"www.lazada.com"
	],
```

File: websites/L/Lazada/metadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"version": "1.0.0",
+	"logo": "https://lzd-img-global.slatic.net/g/tps/tfs/TB1PApewFT7gK0jSZFpXXaTkpXa-200-200.png",
```
Comment (theusaf) [inline websites/L/Lazada/metadata.json:13]: [Logos](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo) should be 512x512px.

File: websites/L/Lazada/presence.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+const presence = new Presence({
+		clientId: "1201520071055515698",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let item: HTMLElement;
```
Comment (theusaf) [inline websites/L/Lazada/presence.ts:6]: It doesn't make sense for this variable to be a global variable considering its usage.

File: websites/L/Lazada/presence.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+			"https://lzd-img-global.slatic.net/g/tps/tfs/TB1PApewFT7gK0jSZFpXXaTkpXa-200-200.png",
+		startTimestamp: browsingTimestamp,	
+	};
+
+	if (document.querySelector(".pdp-mod-product-badge-title")) { // PRODUCT
+		item = document.querySelector(".pdp-mod-product-badge-title");
+		presenceData.details = "Viewing Product:";
+		presenceData.state = item.textContent;
+		presenceData.largeImageKey = document.querySelector('meta[name="og:image"]').getAttribute("content") || "";
+	} else if (document.location.pathname.includes("/tag")) { // TAGS
```
Comment (theusaf) [inline websites/L/Lazada/presence.ts]: To cleanup the multiple uses of `document.location.*`, destructure the used properties:
```ts
const { pathname } = document.location;
...
if (pathname.includes(...)) {
  ...
}
```

File: websites/L/Lazada/presence.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	} else if (document.location.pathname.includes("/user/profile")) {
+		// PROFILE
+		item = document.querySelector(".my-profile-item-info");
+		presenceData.details = "My Profile:";
+		presenceData.state = item.textContent;
+	} else if (document.location.pathname.includes("/address")) {
+		// ADDRESS
+		presenceData.details = "Address Book";
+	} else if (document.location.pathname.includes("/cart")) {
+		// CART
```
Comment (theusaf) [inline websites/L/Lazada/presence.ts:49]: It looks like most of these comments are redundant from the code, so they can be removed

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/L/Lazada/presence.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.ibb.co/LpxnMfj/512x512pngoflazada.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname } = document.location;
+
+	let item: HTMLElement;
```
Comment (Bas950) [inline websites/L/Lazada/presence.ts:13]: Why is the item declared in a let, when it's set in an if case and then only used in the same if case...

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/L/Lazada/presence.ts (hunk 6 above)
Comment (ZzzSleepyy) [reply websites/L/Lazada/presence.ts:13]: I'm using `let` for future changes. Also, I have tried using other variable declaration and it's giving erros.

Comment (theusaf) [reply websites/L/Lazada/presence.ts:13]: You can define variables within scopes such as if statements.
```ts
if (...) {
  const item = ...;
} else if (...) {
  const item = ...;
}
```

File: websites/L/Lazada/presence.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+		item = document.querySelector(".first ");
+		presenceData.details = item.textContent;
+	} else if (pathname.includes("/customer/returns"))
+		presenceData.details = "My Returns";
+	else presenceData.details = "Home";
+
+	if (await presence.getSetting<boolean>("incognito")) {
+		presenceData.details = "Incognito";
+		presenceData.largeImageKey =
+			"https://lzd-img-global.slatic.net/g/tps/tfs/TB1PApewFT7gK0jSZFpXXaTkpXa-200-200.png";
```
Comment (theusaf) [inline websites/L/Lazada/presence.ts:65]:
```suggestion
		presenceData.largeImageKey =
			"https://i.ibb.co/LpxnMfj/512x512pngoflazada.png";
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/L/Lazada/presence.ts (hunk 6 above)
Comment (ZzzSleepyy) [reply websites/L/Lazada/presence.ts]: May I use `document.querySelector(".JrAyI").textContent;` instead?

Comment (theusaf) [reply websites/L/Lazada/presence.ts]: Yes. For that selector specifically, if there is an alternative non computer-generated selector, that would be preferred. If not possible, that's fine too

Comment (ZzzSleepyy) [reply websites/L/Lazada/presence.ts]: I was trying to find a way yesterday, but when I do, I’ll just send another pull request for improvements. Thank you.

File: websites/L/Lazada/presence.ts (hunk 7 above)
Comment (theusaf) [reply websites/L/Lazada/presence.ts:62]: Looks like you accidentally undid this change?

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**